static/js/treatment.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+
+(function () {
+  var translate = Nightscout.language.translate;
+  
+  function initTreatmentDrawer() {
+    $('#eventType').val('BG Check');
+    $('#glucoseValue').val('').attr('placeholder', translate('Value in') + ' ' + browserSettings.units);
+    $('#meter').prop('checked', true);
+    $('#carbsGiven').val('');
+    $('#insulinGiven').val('');
+    $('#preBolus').val(0);
+    $('#notes').val('');
+    $('#enteredBy').val(browserStorage.get('enteredBy') || '');
+    $('#nowtime').prop('checked', true);
+    $('#eventTimeValue').val(moment().format('HH:mm'));
+    $('#eventDateValue').val(moment().format('YYYY-MM-D'));
+  }
+
+  function prepareData() {
+    var data = {
+      enteredBy: $('#enteredBy').val()
+    , eventType: $('#eventType').val()
+    , glucose: $('#glucoseValue').val()
+    , glucoseType: $('#treatment-form').find('input[name=glucoseType]:checked').val()
+    , carbs: $('#carbsGiven').val()
+    , insulin: $('#insulinGiven').val()
+    , preBolus: parseInt($('#preBolus').val())
+    , notes: $('#notes').val()
+    , units: browserSettings.units
+    };
+
+    if ($('#othertime').is(':checked')) {
+      data.eventTime = Nightscout.utils.mergeInputTime($('#eventTimeValue').val(), $('#eventDateValue').val());
+    }
+
+    return data;
+  }
+
+  function treatmentSubmit(event) {
+
+    var data = prepareData();
+
+    confirmPost(data);
+
+    if (event) {
+      event.preventDefault();
+    }
+  }
+
+  function buildConfirmText(data) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/801462997/issues/source?bid=2452374&fileBranchId=2726176#l51)